docs/technical_reference/costing/automate_detailed_unit_costing_rst_file.py

```diff
@@ -0,0 +1,191 @@
+                    f.write("\nCosting Method Variables\n")
+                    f.write("+" * len("Costing Method Variables"))
+                    f.write(
+                        f"\n\nThe following variables are constructed when applying the `cost_{unit_name_list[i]}` costing method in the ``watertap_costing_package``:\n\n"
+                    )
+                    f.write(".. csv-table::\n")
+                    f.write(
+                        '   :header: "Description", "Symbol", "Variable Name", "Default Value", "Units"\n\n'
+                    )
+                    f.write(
+                        '   "description", ":math:`Symbol_{example}`", "variable_name", "1", ":math:`\\text{dimensionless}`"\n'
+                    )
+
+                    # TODO: add capex eqs
+                    f.write("\nCapital Cost Calculations\n")
+                    f.write("+" * len("Capital Cost Calculations"))
+                    f.write(
+                        "\n\nDescribe capital costs..keep it concise where possible\n\n"
+                    )
+                    f.write("    .. math::\n\n")
+                    f.write(
+                        "        C_{cap,tot} = C_{cap,example1}+C_{cap,example2}+C_{cap,other}"
+                    )
+                    f.write("\n\n    .. math::\n\n")
+                    f.write(
+                        "        C_{cap,example1} = fill in equation for each component in total capex equation\n\n "
+                    )
+
+                    # TODO: add opex eqs
+                    f.write("\nOperating Cost Calculations\n")
+                    f.write("+" * len("Operating Cost Calculations"))
+                    f.write(
+                        "\n\nDescribe operating/maintenance costs..keep it concise where possible\n\n"
+                    )
+                    f.write("    .. math::\n\n")
+                    f.write(
+                        "        C_{op,tot} = C_{op,example1}+C_{op,example2}+C_{op,other}"
+                    )
+                    f.write("\n\n    .. math::\n\n")
+                    f.write(
+                        "        C_{op,example1} = fill in equation for each component in total opex equation\n\n "
+                    )
+
+                    # TODO: add module directives to unit and cost method
+                    f.write("\nCode Documentation\n")
+                    f.write("-" * len("Code Documentation"))
+                    f.write(f"\n\n* :mod:`watertap.unit_models.{unit_name_list[i]}`")
+                    f.write(
+                        f"\n* :mod:`watertap.costing.unit_models.{unit_name_list[i]}`"
+                    )
+
+                    f.write("\n\nReferences\n")
+                    f.write("-" * len("References"))
+                    f.write(
+                        "\nAim to include at least one reference in most cases, but delete this section if no references used for cost relationships/default values"
```

Comment:
A disclaimer that the default values of parameters (say unit cost) correspond to the dollar value of the year of the reference (e.g. USD2020). Not necessarily saying this should go in the references, but if supplying units of default parameters should $ be listed as USDYYYY in all of these docs?

Reply:
Perhaps a link to other docs on costing, wherever currency units are explained, with a brief description on the fact that we can resolve variations in defaults (or provided values) with `base_currency`

Reply:
Everything would be converted to base flowsheet currency when solving, so that link probably should exist at a higher level documentation like the `watertap_costing_package` just so it is not unnecessarily duplicated in every unit model. For these docs then, are you agreeing that the parameter tables should be completed using USD_YYYY instead of $ for the currency units?

Reply:
Yes, if reporting default parameters, makes sense to report the currency year.